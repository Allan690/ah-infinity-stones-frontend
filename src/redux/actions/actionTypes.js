export const REQUEST_LOGIN = 'REQUEST_LOGIN';
export const RECEIVE_LOGIN = 'RECEIVE_LOGIN';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const SIGNUP_REQUEST = 'SIGNUP_REQUEST';
export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';

export const SOCIAL_LOGIN_REQUEST = 'SOCIAL_LOGIN_REQUEST';
export const SOCIAL_LOGIN_SUCCESS = 'SOCIAL_LOGIN_SUCCESS';
export const SOCIAL_LOGIN_ERROR = 'SOCIAL_LOGIN_ERROR';


export const REQUEST_INCREMENT = 'REQUEST_INCREMENT';
export const RECEIVE_INCREMENT = 'RECEIVE_INCREMENT';

export const PASSWORD_RESET_START = 'PASSWORD_RESET_START';
export const PASSWORD_RESET_COMPLETE = 'PASSWORD_RESET_COMPLETE';
export const PASSWORD_RESET_ERROR = 'PASSWORD_RESET_ERROR';


export const PASSWORD_RESET_UPDATE_START = 'PASSWORD_RESET_UPDATE_START';
export const PASSWORD_RESET_UPDATE_COMPLETE = 'PASSWORD_RESET_UPDATE_COMPLETE';
export const PASSWORD_RESET_UPDATE_ERROR = 'ASSWORD_RESET_UPDATE_ERROR';
