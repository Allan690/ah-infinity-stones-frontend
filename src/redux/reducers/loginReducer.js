import {
  REQUEST_LOGIN,
  RECEIVE_LOGIN,
  LOGIN_ERROR,
} from '../actions/actionTypes';


const initialState = {
  success: null,
  error: null,
  loading: false,
  isLoggedIn: false,
};

const loginReducer = (state = initialState, action) => {
  switch (action.type) {
    case REQUEST_LOGIN:
      return {
        ...state,
        loading: true,
        error: null,
      };
    case RECEIVE_LOGIN:
      return {
        ...state,
        success: action.data,
        loading: false,
        isLoggedIn: true,
      };
    case LOGIN_ERROR:
      return {
        ...state,
        error: 'Invalid Email or Password',
        loading: false,
        isLoggedIn: false,
      };
    default:
      return state;
  }
};

export default loginReducer;
