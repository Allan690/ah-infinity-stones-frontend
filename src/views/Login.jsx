import React from 'react'

const Login = () => {
  return (
    <div>
      <div className="">
        <h4 className="center">Login</h4>
        <form id="login">
          <label htmlFor="email">Email</label>
          <input type="email" name="email"/>
          <label htmlFor="password">Password</label>
          <input type="password" name="password"/>
          <input type="submit" />
        </form>
      </div>
    </div>
  )
}

export default Login
