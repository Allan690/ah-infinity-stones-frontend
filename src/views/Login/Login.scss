#loginContainer{
width: 40%;
margin-top: 5%;
border-radius: 6px;
border: 1px solid #e6e6e6;
.loader{
    width: 35px;
    height: 35px;
    margin-left: 17.5%;
    position: absolute;
    top: 27%;
    border: 15px solid #f3f3f3;
    border-radius: 50%;
    border-bottom: 16px solid #86C232;
    border-top: 16px solid #222629;
    -webkit-animation: spin 2s linear infinite; 
    animation: spin 2s linear infinite;
    }
    /* Safari browser */
    @-webkit-keyframes spin {
    0% { -webkit-transform: rotate(0deg); }
    100% { -webkit-transform: rotate(360deg); }
  }
  @keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
  }
}
#toast-container{
   margin: 0px auto;
   width: 30%;
   position: absolute;
   top: 28%;
   left: 35%;
}
button.toast-close-button {
    background: 0 0;
    border: 0;
    float: right;
    padding: 0;
    cursor: pointer;
}
