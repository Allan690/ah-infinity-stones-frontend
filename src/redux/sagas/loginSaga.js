/* eslint-disable no-undef */
import { takeLatest, call, put } from 'redux-saga/effects';
import { REQUEST_LOGIN } from '../actions/actionTypes';
import {
  receiveLogin,
  loginError,
} from '../actions/loginActions';
import login from '../../services/login';

/* worker saga */
export function* loginSaga(action) {
  try {
    const { user } = action;
    const response = yield call(login, user);
    if (response.user) {
      if (response.user.Token) {
        yield put(receiveLogin(response.user.Message));
        localStorage.setItem('Token', response.user.Token);
      } else {
        yield put(loginError(response.user.detail));
      }
    } else {
      yield put(loginError(response.errors));
    }
  } catch (error) {
    yield put(loginError(error));
  }
}

/* eslint-disable */
/* watcher saga */
export function* watchLogin() {
    yield takeLatest(REQUEST_LOGIN, loginSaga);
}
