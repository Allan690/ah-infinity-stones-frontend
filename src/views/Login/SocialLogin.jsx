import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import GoogleLogin from 'react-google-login';
import { socialLoginRequest } from '../../redux/actions/socialLoginActions';
import socialConfigs from '../../helpers/socialAuthHelpers';

export class SocialLogin extends PureComponent {
  static propTypes = {
    socialLoginRequest: PropTypes.func.isRequired,
    success: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  };

  static defaultProps = {
    success: null,
  };

  handleTwitter = async (event) => {
    event.preventDefault();
    const { socialLoginRequest: login } = this.props;
    login('twitter');
  }

  handleFacebookLogin = async (event) => {
    event.preventDefault();
    const { socialLoginRequest: login } = this.props;
    login('facebook');
  }

  handleGoogleLogin = (response) => {
    const info = socialConfigs.google(response);
    const { socialLoginRequest: login } = this.props;
    login('google', info);
  };

  handleRedirect = () => { window.location.href = '/'; };

  render() {
    const { success } = this.props;
    return (
      <div className="social-auth">
        { success ? this.handleRedirect() : null }
        <button
          type="button"
          onClick={this.handleFacebookLogin}
          className="kep-login-facebook "
        >
          Facebook
        </button>
        <GoogleLogin
          clientId="31802512181-sh97b417crdhtjtuvabt05chc1e5tn9s.apps.googleusercontent.com"
          buttonText="Google"
          onSuccess={this.handleGoogleLogin}
          onFailure={this.handleGoogleLogin}
        />
        <button
          type="button"
          onClick={this.handleTwitter}
          id="twitter"
        >
          Twitter
        </button>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const { socialLoginReducer: { success } } = state;
  return { success };
};

const mapDispatchToProps = { socialLoginRequest };

export default connect(mapStateToProps, mapDispatchToProps)(SocialLogin);
