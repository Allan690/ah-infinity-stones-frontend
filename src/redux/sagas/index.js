import { all } from 'redux-saga/effects';
import { watchLogin } from './loginSaga';
import watchSignupSaga from './signupSaga';
import watchSocialLogin from './socialLoginSaga';
import { watchSendResetLink, watchPasswordreset } from './passworwordResetSagas';
import {
  watchCreateArticleSaga, watchGetOneArticleSaga,
  watchGetAllArticlesSaga, watchDeleteArticleSaga, watchUpdateArticleSaga,
} from './articleSaga';

export default function* rootSaga() {
  yield all([
    watchSignupSaga(),
    watchLogin(),
    watchSocialLogin(),
    watchSendResetLink(),
    watchPasswordreset(),
    watchCreateArticleSaga(),
    watchUpdateArticleSaga(),
    watchGetOneArticleSaga(),
    watchGetAllArticlesSaga(),
    watchDeleteArticleSaga(),
  ]);
}
