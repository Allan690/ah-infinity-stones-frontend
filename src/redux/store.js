/* eslint no-underscore-dangle: 0 */
/* eslint-disable no-undef */
import { applyMiddleware, createStore, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import allReducers from './reducers';
import rootSaga from './sagas';

// only activate redux devtools on the browser otherwise just compose
const composer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const composeEnhancers = composer;

const middleware = createSagaMiddleware();
const store = createStore(
  allReducers,
  composeEnhancers(applyMiddleware(middleware)),
);

export default store;

middleware.run(rootSaga);
