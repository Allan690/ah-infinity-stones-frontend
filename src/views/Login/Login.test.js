/* global it, expect, mount */
/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Redirect } from 'react-router-dom';
import { call } from 'redux-saga/effects';
import store from '../../redux/store';
import Login from '.';
import loginReducer from '../../redux/reducers/loginReducer';
import * as types from '../../redux/actions/actionTypes';
import { loginSaga } from '../../redux/sagas/loginSaga';
import loginFunc from '../../services/login';
import { handleSuccess, errorClass } from '.';


describe('Login Component', () => {
  let loginContainer;
  let reduxLogin;
  beforeEach(() => {
    loginContainer = shallow(<Login name="login" />);
    reduxLogin = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    );
  });

  it('form renders successfully', () => {
    expect(loginContainer.exists()).toEqual(true);
  });
  it('Should match snapshot', () => {
    expect(loginContainer).toMatchSnapshot();
  });
  it('it renders props correctly', () => {
    const login = shallow(<Login name="login" />);
    expect(login.instance().props.name).toBe('login');
  });
  it('login should render with redux store', () => {
    expect(reduxLogin.exists()).toEqual(true);
  });
  it('it should render a form', () => {
    const form = reduxLogin.find('form');
    expect(form.exists()).toEqual(true);
  });
  it('it should render submit button', () => {
    const button = reduxLogin.find('button');
    expect(button.exists()).toEqual(true);
  });
  it('form should load with 2 input fields', () => {
    const input = reduxLogin.find('input');
    expect(input.length).toEqual(2);
  });
  it('input value is changeable', () => {
    const input = reduxLogin.find('input').at(1);
    input.value = 'hello';
    expect(input.value).toEqual('hello');
  });
});

describe('Login Reducer', () => {
  it('has an initial state', () => {
    expect(loginReducer(undefined, { type: 'unexpected' })).toEqual({
      error: null, loading: false, success: null, isLoggedIn: false,
    });
  });
  it('it can send a login request and update state', () => {
    expect(loginReducer(undefined, {
      type: types.REQUEST_LOGIN,
    })).toEqual({
      error: null, loading: true, success: null, isLoggedIn: false,
    });
  });
});

describe('Login Sagas', () => {
  const userInfo = {
    email: 'test@mail.com',
    password: 'dddddd',
  };
  const generator = loginSaga(userInfo);
  const { user } = userInfo;
  it('Should return user api call', () => {
    expect(generator.next().value).toEqual(call(loginFunc, user));
  });
});

describe('Login makes API fetch', () => {
  const userInfo = {
    email: 'test@mail.com',
    password: 'dddddd',
  };
  it('should return a promise', () => {
    expect(loginFunc()).toBeTruthy();
  });
  it('should return expected data', () => {
    loginFunc(userInfo).then((objects) => {
      expect(objects).toHaveLength(1);
    });
  });
  describe('Login Helpers', () => {
    it('it should redirect', () => {
      const response = Login.handleSuccess('Success');
      expect(response.props.to).toEqual('/');
    });
    it('It should return css error class', () => {
      const response = Login.errorClass('invalid email');
      expect(response).toEqual('is-invalid');
    });
  });
});
