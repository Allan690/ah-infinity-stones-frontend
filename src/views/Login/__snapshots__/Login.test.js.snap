// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component Should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
